CuratorZookeeperClient change-event bug in Unit Test

Generally the Unittests will be successful. But sometimes Unittests failed with an error:
```
[ERROR] Tests run: 12, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 9.731 s <<< FAILURE! - in org.apache.shardingsphere.orchestration.center.instance.CuratorZookeeperCenterRepositoryTest
[ERROR] assertWatchDeletedChangedType(org.apache.shardingsphere.orchestration.center.instance.CuratorZookeeperCenterRepositoryTest)  Time elapsed: 4.995 s  <<< ERROR!
java.util.concurrent.TimeoutException: Timeout waiting for task.
	at org.apache.shardingsphere.orchestration.center.instance.CuratorZookeeperCenterRepositoryTest.assertWatchDeletedChangedType(CuratorZookeeperCenterRepositoryTest.java:105)
```
The details see: 
https://api.travis-ci.org/v3/job/661777933/log.txt  
and the main part of logs: [zk-change.txt](https://github.com/apache/incubator-shardingsphere/files/4336263/zk-change.txt)

This test case use curator testingzkserver and cache to support zk unit test, and then watch a key to fire a delete event.
After comparing with a normal log and an error log.
We find a normal one:
> publish and process a delete event with key `/test/children/5`

And an error one:
> only pushlish a NodeChangeEvent with the parent key `/test/children`, which watched by the last case

So, now we can't reproduce this error by sevaral times running.
But we would like to fix it now and make some further investigation about it：

- [x] set an independent key for each watched test case to avoid only publishing parent event(#4774)
- [ ] test the bug is related of zk-server or curator+cache

--- a/sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepositoryTest.java
+++ b/sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepositoryTest.java
@@ -84,36 +84,36 @@ public final class CuratorZookeeperCenterRepositoryTest {
     
     @Test
     public void assertWatchUpdatedChangedType() throws Exception {
-        centerRepository.persist("/test/children0/1", "value1");
+        centerRepository.persist("/test/children_updated/1", "value1");
         final SettableFuture<DataChangedEvent> future = SettableFuture.create();
-        centerRepository.watch("/test/children0", future::set);
-        centerRepository.persist("/test/children0/1", "value2");
+        centerRepository.watch("/test/children_updated", future::set);
+        centerRepository.persist("/test/children_updated/1", "value2");
         DataChangedEvent dataChangedEvent = future.get(5, TimeUnit.SECONDS);
         assertNotNull(dataChangedEvent);
         assertThat(dataChangedEvent.getChangedType(), is(DataChangedEvent.ChangedType.UPDATED));
-        assertThat(dataChangedEvent.getKey(), is("/test/children0/1"));
+        assertThat(dataChangedEvent.getKey(), is("/test/children_updated/1"));
         assertThat(dataChangedEvent.getValue(), is("value2"));
-        assertThat(centerRepository.get("/test/children0/1"), is("value2"));
+        assertThat(centerRepository.get("/test/children_updated/1"), is("value2"));
     }
     
     @Test
     public void assertWatchDeletedChangedType() throws Exception {
-        centerRepository.persist("/test/children1/5", "value5");
+        centerRepository.persist("/test/children_deleted/5", "value5");
         SettableFuture<DataChangedEvent> future = SettableFuture.create();
-        centerRepository.watch("/test/children1/5", future::set);
-        client.delete().forPath("/test/children1/5");
+        centerRepository.watch("/test/children_deleted/5", future::set);
+        client.delete().forPath("/test/children_deleted/5");
         DataChangedEvent dataChangedEvent = future.get(5, TimeUnit.SECONDS);
         assertNotNull(dataChangedEvent);
         assertThat(dataChangedEvent.getChangedType(), is(DataChangedEvent.ChangedType.DELETED));
-        assertThat(dataChangedEvent.getKey(), is("/test/children1/5"));
+        assertThat(dataChangedEvent.getKey(), is("/test/children_deleted/5"));
         assertThat(dataChangedEvent.getValue(), is("value5"));
     }
     
     @Test
     public void assertWatchAddedChangedType() throws InterruptedException {
-        centerRepository.persist("/test/children2/4", "value4");
+        centerRepository.persist("/test/children_added/4", "value4");
         AtomicReference<DataChangedEvent> actualDataChangedEvent = new AtomicReference<>();
-        centerRepository.watch("/test/children2", actualDataChangedEvent::set);
+        centerRepository.watch("/test/children_added", actualDataChangedEvent::set);
         Thread.sleep(2000L);
         assertNull(actualDataChangedEvent.get());
     }

--- a/sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepositoryTest.java
+++ b/sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-zookeeper-curator/src/test/java/org/apache/shardingsphere/orchestration/center/instance/CuratorZookeeperCenterRepositoryTest.java
@@ -84,36 +84,36 @@ public final class CuratorZookeeperCenterRepositoryTest {
     
     @Test
     public void assertWatchUpdatedChangedType() throws Exception {
-        centerRepository.persist("/test/children/1", "value1");
+        centerRepository.persist("/test/children0/1", "value1");
         final SettableFuture<DataChangedEvent> future = SettableFuture.create();
-        centerRepository.watch("/test/children", future::set);
-        centerRepository.persist("/test/children/1", "value2");
+        centerRepository.watch("/test/children0", future::set);
+        centerRepository.persist("/test/children0/1", "value2");
         DataChangedEvent dataChangedEvent = future.get(5, TimeUnit.SECONDS);
         assertNotNull(dataChangedEvent);
         assertThat(dataChangedEvent.getChangedType(), is(DataChangedEvent.ChangedType.UPDATED));
-        assertThat(dataChangedEvent.getKey(), is("/test/children/1"));
+        assertThat(dataChangedEvent.getKey(), is("/test/children0/1"));
         assertThat(dataChangedEvent.getValue(), is("value2"));
-        assertThat(centerRepository.get("/test/children/1"), is("value2"));
+        assertThat(centerRepository.get("/test/children0/1"), is("value2"));
     }
     
     @Test
     public void assertWatchDeletedChangedType() throws Exception {
-        centerRepository.persist("/test/children/5", "value5");
+        centerRepository.persist("/test/children1/5", "value5");
         SettableFuture<DataChangedEvent> future = SettableFuture.create();
-        centerRepository.watch("/test/children/5", future::set);
-        client.delete().forPath("/test/children/5");
+        centerRepository.watch("/test/children1/5", future::set);
+        client.delete().forPath("/test/children1/5");
         DataChangedEvent dataChangedEvent = future.get(5, TimeUnit.SECONDS);
         assertNotNull(dataChangedEvent);
         assertThat(dataChangedEvent.getChangedType(), is(DataChangedEvent.ChangedType.DELETED));
-        assertThat(dataChangedEvent.getKey(), is("/test/children/5"));
+        assertThat(dataChangedEvent.getKey(), is("/test/children1/5"));
         assertThat(dataChangedEvent.getValue(), is("value5"));
     }
     
     @Test
     public void assertWatchAddedChangedType() throws InterruptedException {
-        centerRepository.persist("/test/children/4", "value4");
+        centerRepository.persist("/test/children2/4", "value4");
         AtomicReference<DataChangedEvent> actualDataChangedEvent = new AtomicReference<>();
-        centerRepository.watch("/test/children", actualDataChangedEvent::set);
+        centerRepository.watch("/test/children2", actualDataChangedEvent::set);
         Thread.sleep(2000L);
         assertNull(actualDataChangedEvent.get());
     }

